lib/perl/Genome/InstrumentData/AlignmentResult/Merged.pm

```diff
@@ -219,24 +226,70 @@ sub create {
         for my $alignment (@alignments) {
             $alignment->add_user(user => $self, label => 'uses');
         }
-
-        return 1;
-    };
-    if (my $error = $@) {
-        $tx->rollback();
-        die $error;
-    } elsif ($rv ne 1) {
+    }
+    catch {
         $tx->rollback();
-        $self->error_message('Unexpected return value: ' . $rv);
-        die $self->error_message;
-    } else {
-        $tx->commit();
+        die $self->error_message('Merge failed due to error: ' . $_);
     }
+    finally {
+        for my $allocation (@temp_allocations) {
+            $allocation->delete;
+        }
+        unless ($_) {
+            $tx->commit();
```

Comment:
The weird part from above, is that multiple `finally` blocks seem to execute in Last In First Out order. So our first `finally` should commit, and our second finally should try to delete everything. I think.